docs/proposals/20220105-observatorium-alerting.md

```diff
@@ -0,0 +1,169 @@
+
+For example, to save alerts we can already PUT the following object to `/api/metrics/v1/{tenant}/api/v1/rules/raw`
+
+```
+groups:
+  - name: test-oidc
+    interval: 5s
+    rules:
+      - alert: HighRequestLatency
+        expr: job:request_latency_seconds:mean5m{job="second"} > 0.5
+        for: 10m
+        labels:
+          team: alpha
+```
+
+Such a rule will be saved through Observatorium API, via Rule-Store to object storage for the specified tenant. It can be then listed through a similar endpoint and GET. Such configuration will be then eventually synchronized with tenant’s Rulers using RuleSync (1m poll by default). This is what was agreed so far and is being tested in downstream RHOBS (Red Hat Observability Service) deployment of Observatorium. 
+
+### ConfigSync
+
+The first decision we propose is to move https://github.com/observatorium/rules-objstore to be pure Go library, potentially in new https://github.com/observatorium/config-client repo. Such library would need to support not only saving and retrieving Rules but also Alerting Routing configuration (and potentially silencer, inhibits and other user configuration in the future). Since alerting routing requires secrets we might need to implement various backends in this client (that's why no `objstore` suffix. ). See [chapter about secrets](#managing-user-secrets) for more details. 
+
+Similarly, we propose to move https://github.com/observatorium/thanos-rule-syncer to https://issues.redhat.com/browse/MON-2129[https://github.com/observatorium/config-sync] that will understand Prometheus Rules configuration, Prometheus Alert Routing configuration and in future silences, inhibitors tpp etc. ConfigSync would then import https://github.com/observatorium/config-objstore. This will allow us to have single focused codebase for all different user configurations we will need. This will allow us to maintain consistency around using it, monitoring, debugging, using secrets etc.
+
+We propose potential https://github.com/observatorium/config-sync to be deployed as a sidecar to each component needing configuration to be present and reloaded dynamically.
+
+As a consequence of this decision I propose Observatorium API to also import https://github.com/observatorium/config-objstore drectly to allow User APIs to manage those items (Rules and Alert routing configuration so far).
+
+### Alert Evaluation
+
+The Ruler itself has to communicate on 4 levels. All connections are presented in the diagram below.
+
+![ruler-connections-diagram](../assets/alerting-ruler.png)
+
+1. Alerts will be evaluated based on periodic requests to any replica of HTTP Query API, potentially through Query-frontend. 
+2. Ruler evaluates Alerts that can be triggered. Triggered alerts will be pushed to all replicas of the Alertmanager (discussed later) through AM HTTP API.
+3. It exposes gRPC Rules API which is consumed by Queriers. Querier than can show this API via it's HTTP Rule API.
```

Comment:
What is meant here by show? Maybe specifying that it's consumed by queriers is enough?